src/Source.h

```diff
@@ -47,6 +47,47 @@ class Source {
     return cur;
   }
 
+  static const char* skipBom(const char* begin) {
+    // Skip any unicode byte order marks
+    switch(begin[0]) {
+      // UTF-32BE
+      case '\x00':
+        if (begin[1] == '\x00' && begin[2] == '\xFE' && begin[3] == '\xFF') {
```

Comment:
Would also be good to include a comment as to the sources of these byte sequences